Source/NSDistributedNotificationCenter.m

```diff
@@ -734,18 +734,39 @@ - (void) _connect
 		@"--auto",
 		nil];
 	    }
-	  [NSTask launchedTaskWithLaunchPath: cmd arguments: args];
-
+#if defined(__MINGW32__)
+          NSTask *task = AUTORELEASE([NSTask new]);
+          [task setStandardError:[NSFileHandle fileHandleForWritingAtPath:@"NUL"]];
+          [task setStandardOutput:[NSFileHandle fileHandleForWritingAtPath:@"NUL"]];
+          [task setLaunchPath:cmd];
+          [task setArguments:args];
+          [task launch];
+#else
+          [NSTask launchedTaskWithLaunchPath: cmd arguments: args];
+#endif
 	  limit = [NSDate dateWithTimeIntervalSinceNow: 10.0];
 	  while (_remote == nil && [limit timeIntervalSinceNow] > 0)
 	    {
               NSAutoreleasePool	*pool = [NSAutoreleasePool new];
 
               [NSThread sleepForTimeInterval: 0.05];
-	      _remote = [NSConnection
-		rootProxyForConnectionWithRegisteredName: service
-		host: host usingNameServer: ns];
-              [_remote retain];
+              NS_DURING
+                {
+                  _remote = [NSConnection
+                              rootProxyForConnectionWithRegisteredName: service
+                                                                  host: host usingNameServer: ns];
+                  RETAIN(_remote);
+                }
+              NS_HANDLER
+                {
+                  NSLog(@"%s:exception: %@", __PRETTY_FUNCTION__, localException);
```

Comment:
Of course we need to remove the commented out code below. But I am also thinking whether __PRETTY_FUNCTION__ will result in a useful method name? Something like NSDebugMLLog would be more useful here.

Reply:
I made the appropriate changes.  Done.